src/hooks/useInterval/useInterval.ts

```diff
@@ -1,26 +1,25 @@
 import { useRef, useEffect } from 'react'
 
-function useInterval(callback: () => void, delay: number | null) {
-  const savedCallback = useRef<() => void | null>()
+export default function useInterval(
+  callback: () => void,
+  delay: number | null,
+) {
+  const savedCallback = useRef(callback)
 
-  // Remember the latest callback.
+  // Remember the latest callback if it changes.
   useEffect(() => {
     savedCallback.current = callback
-  })
+  }, [callback])
```

Comment:
I am wondering if it's the right approach to wrap this in a `useEffect` at all, it might be more performant to simply re-assign this value on every render.